configure-ironic.sh

```diff
@@ -67,15 +63,31 @@ env | grep "^OS_" | tee -a /etc/ironic/ironic.extra
 mkdir -p /shared/html
 mkdir -p /shared/ironic_prometheus_exporter
 
-if [ "$USE_HTTP_BASIC" = "true" ]; then
+HTPASSWD_FILE=/etc/ironic/htpasswd
+if [ -n "${HTTP_BASIC_HTPASSWD}" ]; then
+    printf "%s\n" "${HTTP_BASIC_HTPASSWD}" >"${HTPASSWD_FILE}"
```

Comment:
Do we need an environment variable at all, or could we just say that the thing running the container needs to provide that file? It's possible to mount a secret as a volume, with the contents of the secret exposed as files on the volume.

Reply:
It is possible to mount the secret as a volume, and in fact that's what I'm doing for the client auth below. I think I originally chose an environment variable because I didn't know that you could filter the keys when you mount as a volume (to avoid giving the plaintext password to a server that doesn't need it). You can, although it would be a little weird because you'd have to define two volumes on the Pod for each secret (with different lists of keys), and then mount different ones in different places in different Containers.

The HTTP_BASIC_HTPASSWD only contains a hash, so there's no security driver for mounting it as a volume - it's safe to pass as an environment variable, write to disk &c. So I think it could go either way. Is there a reason to prefer volume vs environment variable?

Reply:
I just thought that instead of the container writing the contents of a variable to disk, the file could be provided directly. It's not a big deal, and maybe the consistency of always using environment variables is better.